runtime/agent/CMakeLists.txt

```diff
@@ -20,5 +25,24 @@ target_link_libraries(bpftime-agent
   -ldl
   spdlog::spdlog
   bpftime_frida_uprobe_attach_impl
-  bpftime_syscall_trace_attach_impl
 )
+
+else()
+target_include_directories(bpftime-agent
+  PRIVATE
+  ${FRIDA_GUM_INSTALL_DIR}
+  ../include
+  ../../third_party/
+  ${SPDLOG_INCLUDE}
+  ${FRIDA_UPROBE_ATTACH_IMPL_INCLUDE}
```

Comment:
Shouldn't we keep `SYSCALL_TRACE_ATTACH_IMPL_INCLUDE`?

Reply:
> Shouldn't we keep `SYSCALL_TRACE_ATTACH_IMPL_INCLUDE`?

It's not supported on platforms other than x86_64

Reply:
Can't we support it for ARM?

Reply:
> Can't we support it for ARM?

We used a mechanism that was only designed for x86, in file https://github.com/eunomia-bpf/bpftime/blob/master/attach/text_segment_transformer/text_segment_transformer.cpp . We replace instruction `syscall` to `call *%rax` (both of them are two bytes) and setup appropriate trampoline code on page zero to implement syscall trace on x86. If we want to support syscall trace on other platforms, we need a new mechanism to implement similar thing

Reply:
discussion out of scope this PR. Resolving it, cc: @Officeyutong @viniciusd